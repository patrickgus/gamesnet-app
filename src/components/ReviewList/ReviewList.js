import React, { Component } from "react";
import ReviewListItem from "../ReviewListItem/ReviewListItem";
import { Section } from "../../components/Utils/Utils";
import reviewList from "../../stores/review-store";

export default class ReviewList extends Component {
  renderReviews() {
    return reviewList.map(review => (
      <ReviewListItem key={review.id} review={review} />
    ));
  }

  render() {
    return (
      <Section list className="ReviewList">
        {this.renderReviews()}
      </Section>
    );
  }
}
