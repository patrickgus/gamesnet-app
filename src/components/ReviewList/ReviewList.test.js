import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ReviewList from "./ReviewList";

describe("ReviewList component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ReviewList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the list as expected", () => {
    const tree = renderer.create(<ReviewList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
