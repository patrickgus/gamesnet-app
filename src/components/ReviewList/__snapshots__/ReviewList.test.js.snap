// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList component renders the list as expected 1`] = `
<section
  className="Section Section--list ReviewList"
>
  <section
    className="Section ReviewListItem"
  >
    <h4>
      "
      The greatest game of all time
      "
    </h4>
    <p>
      Rating: 
      10
    </p>
    <div
      className="reviewInfo"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fermentum lorem malesuada, posuere felis sed, ultrices orci. Phasellus varius neque eget lobortis mollis. Cras non venenatis nibh. Maecenas in nulla ut tellus dictum mollis. Maecenas libero velit, accumsan et ultrices vel, dapibus venenatis nibh. Sed nisi tellus, ullamcorper non neque eu, pharetra mattis risus. Sed metus ipsum, dignissim nec risus eget, pellentesque tincidunt enim. Quisque dui urna, pellentesque eu odio eget, imperdiet rutrum magna.
      </p>
    </div>
  </section>
  <section
    className="Section ReviewListItem"
  >
    <h4>
      "
      My favorite game
      "
    </h4>
    <p>
      Rating: 
      9
    </p>
    <div
      className="reviewInfo"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fermentum lorem malesuada, posuere felis sed, ultrices orci. Phasellus varius neque eget lobortis mollis. Cras non venenatis nibh. Maecenas in nulla ut tellus dictum mollis. Maecenas libero velit, accumsan et ultrices vel, dapibus venenatis nibh. Sed nisi tellus, ullamcorper non neque eu, pharetra mattis risus. Sed metus ipsum, dignissim nec risus eget, pellentesque tincidunt enim. Quisque dui urna, pellentesque eu odio eget, imperdiet rutrum magna.
      </p>
    </div>
  </section>
  <section
    className="Section ReviewListItem"
  >
    <h4>
      "
      Simply, the best
      "
    </h4>
    <p>
      Rating: 
      10
    </p>
    <div
      className="reviewInfo"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fermentum lorem malesuada, posuere felis sed, ultrices orci. Phasellus varius neque eget lobortis mollis. Cras non venenatis nibh. Maecenas in nulla ut tellus dictum mollis. Maecenas libero velit, accumsan et ultrices vel, dapibus venenatis nibh. Sed nisi tellus, ullamcorper non neque eu, pharetra mattis risus. Sed metus ipsum, dignissim nec risus eget, pellentesque tincidunt enim. Quisque dui urna, pellentesque eu odio eget, imperdiet rutrum magna.
      </p>
    </div>
  </section>
</section>
`;
